go-controller/pkg/metrics/master.go

```diff
@@ -246,37 +259,24 @@ func RegisterMasterMetrics(nbClient, sbClient goovn.Client) {
 		prometheus.MustRegister(metricV6AllocatedHostSubnetCount)
 		prometheus.MustRegister(metricEgressIPCount)
 		prometheus.MustRegister(metricEgressFirewallRuleCount)
+		prometheus.MustRegister(metricIPsecEnabled)
 		registerWorkqueueMetrics(MetricOvnkubeNamespace, MetricOvnkubeSubsystemMaster)
 	})
 }
 
-// StartE2ETimeStampMetricUpdater adds a goroutine that updates a "timestamp" value in the
-// nbdb every 30 seconds. This is so we can determine freshness of the database
-func StartE2ETimeStampMetricUpdater(stopChan <-chan struct{}, ovnNBClient goovn.Client) {
-	startE2ETimeStampUpdaterOnce.Do(func() {
+// StartMasterMetricUpdater adds a goroutine that updates a "timestamp" value in the
+// nbdb every 30 seconds. This is so we can determine freshness of the database.
+// Also, update IPsec enabled or disable metric.
+func StartMasterMetricUpdater(stopChan <-chan struct{}, nbClient client.Client) {
+	startMasterMetricUpdaterOnce.Do(func() {
+		addIPSecMetricHandler(nbClient)
 		go func() {
-			tsUpdateTicker := time.NewTicker(30 * time.Second)
+			ticker := time.NewTicker(30 * time.Second)
+			defer ticker.Stop()
```

Comment:
Nit: Do you need this manual stop? Shouldn't the ticker stop once the go routine returns?

Reply:
Yeah, its pointless but I did it anyway because its recommended by the time package docs. Once the go routine returns, the memory allocated within that context is free'd I believe.